jobbergate-cli/jobbergate_cli/subapps/tools.py

```diff
@@ -0,0 +1,52 @@
+"""
+Tools for the subapps.
+"""
+
+from jobbergate_cli.exceptions import Abort
+
+
+def sanitize_application_selection(
+    id_or_identifier: int | str | None = None, id: int | None = None, identifier: str | None = None, prefix: str = ""
+) -> int | str:
+    """
+    Sanitize the application selection parameters.
+    """
+    counter = sum((0 if i is None else 1 for i in (id_or_identifier, id, identifier)))
+    prefix_underline = prefix.replace("-", "_") + "_"
+    prefix_dash = prefix.replace("_", "-") + "-"
+
+    if counter != 1:
+        raise Abort(
+            (
+                f"You must supply one and only one selection value "
+                f"(positional [yellow]{prefix_underline}id_or_identifier[/yellow], "
+                f"[yellow]--{prefix_dash}id[/yellow], or [yellow]{prefix_dash}identifier[/yellow]), got {counter}"
+            ),
+            subject="Invalid Selection",
+            support=False,
+        )
+    if isinstance(id_or_identifier, str) and id_or_identifier.isdigit():
+        id_or_identifier = int(id_or_identifier)
+
+    return id_or_identifier or id or identifier  # type: ignore
```

Comment:
can you add a comment on why this is ignored?

Reply:
I think it would be good to resolve the type here.

Reply:
Addressed by https://github.com/omnivector-solutions/jobbergate/pull/636/commits/d172ceeefc16fa4bf60aa99592ed00804a4d4315. I changed a little the syntax to clarify the types no mypy, so now it is happy.